docs/docs/development/hardware-integration/dongle.mdx

```diff
@@ -0,0 +1,301 @@
+        >;
+    };
+```
+
+Copy this node into your `my_keyboard_dongle.overlay` file. Take care to put it under the root node (the `/ { \* Stuff *\ };` node).
+If there are multiple matrix transformations, copy the one that matches your keyboard's layout.
+
+Make a note of the label that the transform has, it will be used later. In the example the label is `default_transform`.
+
+#### Physical layout
+
+A full physical layout is necessary to allow your dongle to be used with [ZMK Studio](../../features/studio.md).
+If your keyboard is not Studio-ready or you have no interest in using ZMK Studio with your dongle, then this section is simplified significantly.
+
+<Tabs
+    groupId="studio-ready"
+    defaultValue="studio"
+    values={[
+      { label: "Dongle with Studio", value: "studio" },
+      { label: "Dongle without Studio", value: "nostudio" }
+    ]}
+  >
+<TabItem value="studio">
+
+You will need to find your keyboard's physical layout, much like finding the matrix transform.
+Depending on your keyboard, the physical layout could be found in a variety of locations. Look for a node with `compatible = "zmk,physical-layout";`.
+There are three commonly found possibilities:
+
+- The physical layout is found in a file called `my_keyboard-layouts.dtsi`
+- The physical layout is found in the same file as the matrix transform and chosen node
+- The physical layout is imported from ZMK's shared layouts - `#include <layouts/<stuff>.dtsi>` can be found at the top of one of the devicetree files.
+
+<Tabs
+    groupId="layout-locations"
+    defaultValue="layouts"
+    values={[
+      { label: "Layouts File", value: "layouts" },
+      { label: "Base File", value: "base" },
+      { label: "Imported", value: "imported" }
+    ]}
+  >
+<TabItem value="layouts">
+Copy the file into your dongle's folder. Then import the file, assign the matrix transform to it, and select it in the `chosen` node:
+
+```dts
+#import "my_keyboard-layouts.dtsi"
+
+&physical_layout0 {
+    transform = <&default_transform>;
+};
+
+/ {
+    chosen {
+        zmk,kscan = &mock_kscan;
+        zmk,physical-layout = &physical_layout0;
+    };
+};
+```
+
+Make sure that the labels `physical_layout0` and `default_transform` match those of the physical layout node and the matrix transform respectively.
+
+If there are multiple physical layouts in the file, you will need to copy over all of the remaining matrix transformations and assign them similarly.
+
+</TabItem>
+<TabItem value="base">
+Copy the physical layout node into your `my_keyboard_dongle.overlay` file. Make sure the matrix transform is assigned to it, and select it in the `chosen` node.
+
+```dts
+/ {
+    chosen {
+        zmk,kscan = &mock_kscan;
+        zmk,physical-layout = &physical_layout0;
+    };
+
+    physical_layout0: physical_layout_0 {
+        compatible = "zmk,physical-layout";
+        display-name = "Default Layout";
+        transform = <&default_transform>;
+        keys = //Long list of keys
```

Comment:
```suggestion
        keys = <...>; // Long list of key positions
```
This is similar to how it is used in https://zmk.dev/docs/development/hardware-integration/physical-layouts#multiple-physical-layouts